pkg/codegen/schema/pulumi.json

```diff
@@ -526,6 +532,143 @@
+            "description": "Describes NodeJS specific settings",
+            "type": "object",
+            "properties": {
+                "packageName": {
+                    "description": "NPM package name (includes @namespace)",
+                    "type": "string"
+                },
+                "packageDescription": {
+                    "description": "NPM package description",
+                    "type": "string"
+                },
+                "readme": {
+                    "description": "Content of the generated package README.md file.",
+                    "type": "string"
+                },
+                "compatibility": {
+                    "description": "",
+                    "type": "string",
+                    "enum": ["tfbridge20"]
+                },
+                "dependencies": {
+                    "description": "NPM package dependencies",
+                    "$ref": "#/$defs/schemaStringMap"
+                },
+                "devDependencies": {
+                    "description": "NPM package devDependencies",
+                    "$ref": "#/$defs/schemaStringMap"
+                },
+                "disableUnionOutputTypes": {
+                    "description": "",
+                    "type": "boolean"
+                },
+                "typescriptVersion": {
+                    "description": "Typescript Version",
+                    "type": "string"
+                }
+            }
+        },
+        "pythonLanguageSpec": {
+            "title": "Python Language Overrides Definition",
+            "description": "Describes Python specific settings",
+            "type": "object",
+            "properties": {
+                "packageName": {
+                    "description": "NPM package name (includes @namespace)",
+                    "type": "string"
+                },
+                "readme": {
+                    "description": "Content of the generated package README.md file.",
+                    "type": "string"
+                },
+                "compatibility": {
+                    "description": "",
+                    "type": "string",
+                    "enum": ["tfbridge20"]
+                },
+                "requires": {
+                    "description": "Python package dependencies for the generated Python SDK",
+                    "$ref": "#/$defs/schemaStringMap"
+                }
+            }
+        },
+        "javaLanguageSpec": {
+            "title": "Java Language Overrides Definition",
+            "description": "Describes Java specific settings",
+            "type": "object",
+            "properties": {
+                "basePackage": {
+                    "description": "Base Java package name for the generated Java provider SDK",
+                    "type": "string",
+                    "default": "com.pulumi"
+                }
+            }
+        },
+        "languageSpec": {
```

Comment:
Curious, you'd need to link languageSpec somewhere right? For it to be used to describe the type of .language field?

Reply:
I documented how to have JSON schema validation when manually writing the Pulumi schema file here:

https://github.com/pulumi/pulumi-provider-boilerplate/tree/dd69ad8e42d60334aa7026e8d790a7805aefddb2#writing-the-schema

But it seems this was removed from the README in the past 2 days by @dixler. I also don't know how much of the JSON schema file is used within engineering to validate if the contents of Pulumi schema files are correct.